File: swift/0380-insert-delete-getrandom-o1.swift
class RandomizedSet {
    var random = Set<Int>()

    init() {
    
    }
    
    func insert(_ val: Int) -> Bool {
        if !random.contains(val){
            random.insert(val)
            return true
        } else {
            return false
        }
    }
    
    func remove(_ val: Int) -> Bool {
        if random.contains(val) {
            random.remove(val)
            return true
        } else {
            return false
        }
    }
    
    func getRandom() -> Int {
        return random.randomElement()!
    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * let obj = RandomizedSet()
 * let ret_1: Bool = obj.insert(val)
 * let ret_2: Bool = obj.remove(val)
 * let ret_3: Int = obj.getRandom()
 */

File: kotlin/0380-insert-delete-getrandom-o1.kt
/*
* Containing the logic behind the operations, but a cleaner Kotlin solution is provided below.
* Here implement some of the logic of the functions ourselves, as in the video, to have this solution compatible with the video solution.
*/
class RandomizedSet() {

    val hs = HashSet<Int>()

    fun insert(`val`: Int): Boolean {
        if(hs.contains(`val`)) 
            return false
        else {
            hs.add(`val`)
            return true
        }
    }

    fun remove(`val`: Int): Boolean {
        if(hs.contains(`val`)){
            hs.remove(`val`)
            return true
        }else 
            return false
    }

    fun getRandom(): Int {
        return hs.random()
    }

}
 
/*
* Cleaner Kotlin solution. add() and remove()
*/
class RandomizedSet() {

    val hs = HashSet<Int>()

    fun insert(`val`: Int) = hs.add(`val`)

    fun remove(`val`: Int) = hs.remove(`val`)

    fun getRandom() = hs.random()

}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * var obj = RandomizedSet()
 * var param_1 = obj.insert(`val`)
 * var param_2 = obj.remove(`val`)
 * var param_3 = obj.getRandom()
 */
